src/utils/stateUpdateUtils.tsx

```diff
@@ -0,0 +1,120 @@
+import _ from 'lodash';
+import httpClient from '../httpClient'
+import { AggregationFactor, Filters, FilterType } from '../types'
+
+export const updateFilters = async (
+  dispatch: any,
+  filters: Filters,
+  filterType: FilterType,
+  filterValue: any[],
+  exploreIsOpen: Boolean,
+  page: string,
+  aggregationFactor: AggregationFactor) => {
+
+  dispatch({
+    type: "CHANGE_LOADING",
+    payload: {
+      pageStateEnum: page,
+      isLoading: true
+    }
+  })
+
+  dispatch({
+    type: 'UPDATE_FILTER',
+    payload: {
+      filterType,
+      filterValue,
+      pageStateEnum: page
+    }
+  });
+
+  const updatedFilters: Filters = Object.assign({}, filters);
+  updatedFilters[filterType] = new Set(filterValue);
+
+  const api = new httpClient()
+  // Update current records when called
+  const [records,
+    estimateGradePrevalences, reAggregatedRecords] = await
+      Promise.all([
+        api.getAirtableRecords(filters, exploreIsOpen),
+        api.getEstimateGrades(filters),

```

Comment:
we only need estimate grades when page === "explore" right?

Reply:
estimate grades are used for the summary stats